bonan_richards_eqn/get_bonan_data.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# This script downloads a specific version of a Matlab script from a GitHub repository
+# and generates data files for two different soil types using that script.
+# The output data files are saved in the paths bonanmodeling/sp_08_01/bonan_data_clay.txt
+# and bonanmodeling/sp_08_01/bonan_data_sand.txt, relative to the directory
+# where this script is run.
+
+# Parse the input argument for the output directory
+output_dir=$1
+
+# Clone the repository containing the Matlab script we need
+git clone https://github.com/gbonan/bonanmodeling.git
+cd bonanmodeling
+
+# Checkout the commit that we know has what we need
+git -c advice.detachedHEAD=false checkout a10cf764013be58c2def1dbe7c7e52a3213e061e
+
+# Comment out Bonan's Haverkamp formulation for hydraulic conductivity (K)
+# This will use the van Genuchten formulation instead, as is done in ClimaLand.jl
+awk '{if (NR>33&&NR<42) $0 = "%" substr($0, 2); print}' sp_08_01/van_Genuchten.m > tmpfile && mv tmpfile sp_08_01/van_Genuchten.m
+
+# Load the Matlab module on Caltech's Central cluster
+# NOTE: You will need to change it if running on a different machine
+module load matlab/r2024a
+
+# Run the Matlab script to generate the data for the sand soil type
+matlab -batch "run('sp_08_01/sp_08_01.m'); exit;"
+
+# Remove the header (first line) and rename the generated data file
+tail -n +2 sp_08_01/data1.txt > ../$output_dir/bonan_data_sand.txt
```

Comment:
I think this will fail if $output_dir has a space. It probably should be
```suggestion
tail -n +2 sp_08_01/data1.txt > "../$output_dir/bonan_data_sand.txt"
```
same below